### PR TITLE
Add QP middleware to auto-parse string values to correct type (backported from master)

### LINKED ISSUE #13263: Clicking ID field in Table view shows error
**Describe the bug**
When clicking on a primary key field in the table view of a query, the detail view of the record should be opened. Instead an error message "There was a problem with your question" with the postgres error `ERROR: operator does not exist: bigint = character varying Hint: No operator matches the given name and argument type(s). You might need to add explicit type casts. Position: 1323` is shown.

Showing a detail page works by adding a filter to the custom query with the given ID manually.

**Logs**
[metabase logs.txt](https://github.com/metabase/metabase/files/5238706/metabase.logs.txt)

**To Reproduce**
1. Create a new custom question
2. Select a table with a DB id, the id should be shown as a link to the details page for the record
3. Click the link

**Expected behavior**
Clicking on a primary key should open the details page and show the selected record.

**Screenshots**
grafik
grafik

**Information about your Metabase Installation:**

```json
{
  "browser-info": {
    "language": "de",
    "platform": "MacIntel",
    "userAgent": "Mozilla/5.0 (Macintosh; Intel Mac OS X 10.15; rv:81.0) Gecko/20100101 Firefox/81.0",
    "vendor": ""
  },
  "system-info": {
    "file.encoding": "UTF-8",
    "java.runtime.name": "OpenJDK Runtime Environment",
    "java.runtime.version": "11.0.8+10",
    "java.vendor": "AdoptOpenJDK",
    "java.vendor.url": "https://adoptopenjdk.net/",
    "java.version": "11.0.8",
    "java.vm.name": "OpenJDK 64-Bit Server VM",
    "java.vm.version": "11.0.8+10",
    "os.name": "Linux",
    "os.version": "4.4.0-165-generic",
    "user.language": "en",
    "user.timezone": "GMT"
  },
  "metabase-info": {
    "databases": [
      "mysql",
      "postgres"
    ],
    "hosting-env": "unknown",
    "application-database": "postgres",
    "application-database-details": {
      "database": {
        "name": "PostgreSQL",
        "version": "12.4 (Debian 12.4-1.pgdg100+1)"
      },
      "jdbc-driver": {
        "name": "PostgreSQL JDBC Driver",
        "version": "42.2.8"
      }
    },
    "run-mode": "prod",
    "version": {
      "date": "2020-09-15",
      "tag": "v0.36.6",
      "branch": "release-0.36.x",
      "hash": "cb258fb"
    },
    "settings": {
      "report-timezone": "Europe/Berlin"
    }
  }
}
```

**Severity**
It is this annoying, blocking an upgrade.
Predefined questions/dashboards still work but it's not possible to browse through data anymore.

**Additional context**
The `ID` columns are set as "Entity Key". The problem is for all tables.

### PR DESCRIPTION
**Pull request by camsaul**
Fixes #13263

I added this middleware as part of #13354 to get the new endpoints working. This PR backports that middleware from `master` as well as the Postgres Cypress tests from #13311

### CODE REVIEW

**Comment by robdaemon** [comment]
This might help some of the JS id-to-string stuff we've got. One note, I don't see any code changes, just a Cypress test in this PR?

**Comment by camsaul** [comment]
@robdaemon good catch, I guess I forgot to cherry-pick the code in question

File: src/metabase/query_processor/middleware/auto_parse_filter_values.clj [hunk 1]
```diff
@@ -0,0 +1,44 @@
...
+             [i18n :refer [tru]]
+             [schema :as su]]
+            [schema.core :as s]))
+
+(s/defn ^:private parse-value-for-base-type [v :- s/Str, base-type :- su/FieldType]
+  {:pre [(string? v)]}
+  (try
+    (condp #(isa? %2 %1) base-type
+      :type/BigInteger (bigint v)
+      :type/Integer    (Integer/parseInt v)
```
Comment (robdaemon) [inline src/metabase/query_processor/middleware/auto_parse_filter_values.clj:20]: I don't see `Long` in here, what happens for >32 bit integers?

Comment (camsaul) [reply src/metabase/query_processor/middleware/auto_parse_filter_values.clj]: Good point. Fixed

Comment (robdaemon) [reply src/metabase/query_processor/middleware/auto_parse_filter_values.clj]: without looking, I bet we do this elsewhere

**Review by robdaemon — APPROVED** [review]

File: src/metabase/query_processor/middleware/auto_parse_filter_values.clj (hunk 1 above)
Comment (camsaul) [reply src/metabase/query_processor/middleware/auto_parse_filter_values.clj]: did a quick grep of the codebase and couldn't find it anywhere else. We are using `Long/parseLong` in 4 other places tho.

**Pull request merged**